spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentDeleteServiceInstanceWorkflow.java

```diff
@@ -75,34 +77,52 @@ public AppDeploymentDeleteServiceInstanceWorkflow(BrokeredServices brokeredServi
 	}
 
 	private Flux<String> deleteBackingServices(DeleteServiceInstanceRequest request) {
-		return collectConfiguredBackingServices(request)
-			.mergeWith(collectBoundBackingServices(request))
-			.doOnEach(backingServices -> log.debug("Deleting backing services {} for {}/{}",
-				backingServices, request.getServiceDefinition().getName(), request.getPlan().getName()))
-			.flatMap(backingServicesProvisionService::deleteServiceInstance)
+		return collectBackingServices(request)
+			.collectList()
+			.flatMapMany(backingServices -> {
+				if (!CollectionUtils.isEmpty(backingServices)) {
+					return backingServicesProvisionService.deleteServiceInstance(backingServices);
+				}
+				return Flux.empty();
```

Comment:
`Flux.empty().collectList()` will result in an empty list instead of null, so I feel this null check here is not necessary.

Reply:
I got a NPE in the mapper without that check. I agree it's ugly and feels unnecessary. :-/